hyperspy/_signals/eds.py

```diff
@@ -122,7 +122,7 @@ def _get_beam_energy(self):
         units_name = self.axes_manager.signal_axes[0].units
 
         if units_name == 'eV':
-            beam_energy = beam_energy * 1000
+            beam_energy *= 1000
         return beam_energy
```

Comment:
Is it possible for `beam_energy` to be a reference (like an `ndarray`)? If so, this change might mean that you will change the value stored in the metadata: Previously the value of `beam_energy` was copied and reassigned, now it modifies the value "in place".